event-session-signup/Code.gs

```diff
@@ -0,0 +1,176 @@
+    var end = joinDateAndTime_(session[1], session[3]);
+    var options = {location: session[4], sendInvites: true};
+    var event = cal.createEvent(title, start, end, options)
+        .setGuestsCanSeeGuests(false);
+    session[5] = event.getId();
+  }
+  range.setValues(values);
+
+  // Store the ID for the Calendar, which is needed to retrieve events by ID.
+  ScriptProperties.setProperty('calId', cal.getId());
+}
+
+/**
+ * Creates a single Date object from separate date and time cells.
+ *
+ * @param {Date} date A Date object from which to extract the date.
+ * @param {Date} time A Date object from which to extract the time.
+ * @return {Date} A Date object representing the combined date and time.
+ */
+function joinDateAndTime_(date, time) {
+  date = new Date(date);
+  date.setHours(time.getHours());
+  date.setMinutes(time.getMinutes());
+  return date;
+}
+
+/**
+ * Creates a Google Form that allows respondents to select which conference
+ * sessions they would like to attend, grouped by date and start time.
+ *
+ * @param {Spreadsheet} ss The spreadsheet that contains the conference data.
+ * @param {Array<String[]>} values Cell values for the spreadsheet range.
+ */
+function setUpForm_(ss, values) {
+  // Group the sessions by date and time so that they can be passed to the form.
+  var schedule = {};
+  for (var i = 1; i < values.length; i++) {
+    var session = values[i];
+    var day = session[1].toLocaleDateString();
+    var time = session[2].toLocaleTimeString();
+    if (!schedule[day]) {
+      schedule[day] = {};
+    }
+    if (!schedule[day][time]) {
+      schedule[day][time] = [];
+    }
+    schedule[day][time].push(session[0]);
+  }
+
+  // Create the form and add a multiple-choice question for each timeslot.
+  var form = FormApp.create('Conference Form');
+  form.setDestination(FormApp.DestinationType.SPREADSHEET, ss.getId());
+  form.addTextItem().setTitle('Name').setRequired(true);
+  form.addTextItem().setTitle('Email').setRequired(true);
+  for (var day in schedule) {
+    var header = form.addSectionHeaderItem().setTitle('Sessions for ' + day);
+    for (var time in schedule[day]) {
+      var item = form.addMultipleChoiceItem().setTitle(time + ' ' + day)
+          .setChoiceValues(schedule[day][time]);
+    }
+  }
+}
+
+/**
+ * A trigger-driven function that sends out calendar invitations and a
+ * personalized Google Docs itinerary after a user responds to the form.
+ *
+ * @param {Object} e The event parameter for form submission to a spreadsheet;
+ *     see https://developers.google.com/apps-script/understanding_events
+ */
+function onFormSubmit(e) {
+  var user = {name: e.namedValues['Name'][0], email: e.namedValues['Email'][0]};
+
+  // Grab the session data again so that we can match it to the user's choices.
+  var response = [];
+  var values = SpreadsheetApp.getActive().getSheetByName('Conference Setup')
+     .getDataRange().getValues();
+  for (var i = 1; i < values.length; i++) {
```

Comment:
Depending on how other 2 cases are handled, maybe just a quick comment explaining row 0 is skipped because it's a header.